File: Easy/Math/3754. Concatenate Non-Zero Digits and Multiply by Sum I.cpp
// 3754. Concatenate Non-Zero Digits and Multiply by Sum I.py
class Solution {
public:
    long long sumAndMultiply(int n) {
        if(n == 0) return 0;
        string s = to_string(n);
        long long sum = 0;
        string temp = "";
        for(auto i : s){
            if(i != '0')temp+=i;
            sum+= i - '0';
        }
        int x = stoi(temp);
        sum = sum*x;
        return sum;
        
    }
};

File: Easy/Math/3754. Concatenate Non-Zero Digits and Multiply by Sum I.py
# 3754. Concatenate Non-Zero Digits and Multiply by Sum I
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x = 0
        base = 1
        sum = 0

        while(n > 0):
            rem = n%10
            x+= (rem*base)
            if(rem != 0): base *= 10
            n//=10
            sum+=rem
        return x*sum

    
